mix.exs

```diff
@@ -123,6 +128,54 @@ defmodule Sentry.Mixfile do
   end
 
   defp aliases do
-    [test: ["sentry.package_source_code", "test"]]
+    [
+      test: ["sentry.package_source_code", "test", &run_integration_tests_if_supported/1],
+      "test.integrations": &test_integrations/1
+    ]
+  end
+
+  defp run_integration_tests_if_supported(_args) do
+    if Version.match?(System.version(), ">= 1.16.0") do
+      Mix.Task.run("test.integrations")
+    else
+      IO.puts("Skipping integration tests for Elixir versions < 1.16")
+    end
+  end
+
+  @integrations [
+    "phoenix_app"
+  ]
+
+  defp test_integrations(args) do
+    if Version.match?(System.version(), ">= 1.16.0") do
+      for integration <- @integrations do
+        run_integration_tests(integration, args)
+      end
+    else
+      IO.puts("Skipping integration tests for Elixir versions < 1.16")
+    end
+  end
+
+  defp run_integration_tests(integration, args) do
+    header = if IO.ANSI.enabled?(), do: IO.ANSI.bright() <> IO.ANSI.cyan(), else: ""
+    reset = if IO.ANSI.enabled?(), do: IO.ANSI.reset(), else: ""
+
+    IO.puts("\n#{header}==> Running tests for integration: #{integration}#{reset}")
+
+    integration_dir = Path.join("test_integrations", integration)
+
+    {_, status} =
+      System.cmd("sh", ["-c", "cd #{integration_dir} && mix test #{Enum.join(args, " ")}"],
+        into: IO.binstream(:stdio, :line)
+      )
```

Comment:
Instead of `cd <> && ...`, you can use the `:cd` option in `System.cmd` and you won't have to use `sh`:

```suggestion
    {_, status} =
      System.cmd("mix", ["test" | args],
        into: IO.binstream(:stdio, :line),
        cd: Path.join("test_integrations", integration)
      )
```

Reply:
Also, this snippet is from another repo I have and it's handy to show nice test output when running in a subshell, cause it defaults to no colors:

```elixir
color_arg = if IO.ANSI.enabled?(), do: "--color", else: "--no-color"

System.cmd("mix", ["test", color_arg | args], ...)
```